src/document_options_save_as_json.lua

```diff
@@ -0,0 +1,709 @@
+        TipWidth = normalize_funcs.d10k,
+        ["tRelativeInset"] = normalize_funcs.m100
+    },
+    FCTupletPrefs = {
+        BracketThickness = normalize_funcs.d64,
+        MaxSlope = normalize_funcs.d10
+    }    
+}
+
+function normalize_units_for_table(t, pattern, func)
+    for key, value in pairs(t) do
+        if type(value) == "table" then 
+            normalize_units_for_table(value, pattern, func)
+        elseif string.find(key, pattern) then
+            t[key] = func(value)
+        end
+    end
+end    
+
+function normalize_units_for_section(section_table, section_tag)
+    local selectors = norm_func_selectors[section_tag]
+    if selectors then
+        for pattern, func in pairs(selectors) do            
+            normalize_units_for_table(section_table, pattern, func)
+        end
+    end
+end
+
+-- endregion
+
+function get_last_key(t)
+    local result
+    for k, _ in pairsbykeys(t) do result = k end
+    return result
+end
+
+-- Simple implementation because we know our inputs are simple
+-- Can't use a json library because we need to keep keys in order
+function get_as_json(t, indent)
+    indent = indent or 0
+    local result = {}
+
+    table.insert(result, '{\n')
+    indent = indent + 1
+    local spaces = string.rep(" ", indent * 2) 
+
+    local last_key = get_last_key(t)
+    for key, val in pairsbykeys(t) do
+        local maybe_comma = key ~= last_key and ',' or ''
+        if type(val) == "table" then 
+            table.insert(result, string.format('%s"%s": ', spaces, key))
+            table.insert(result, get_as_json(val, indent))
+            table.insert(result, string.format("%s}%s\n",
+                spaces,
+                maybe_comma
+            ))
+        else 
+            table.insert(result, string.format('%s"%s": %s%s\n',
+                spaces, 
+                key, 
+                type(val) == "string" and string.format('"%s"', val) or tostring(val),
+                maybe_comma
+            ))
+        end
+    end
+
+    if indent == 1 then
+        table.insert(result, "}")
+    end 
+    return table.concat(result)
+end
+
+function insert_header(prefs_table, document)
+    local file_path = finale.FCString()
+    document:GetPath(file_path)
+    local key = "@Meta"
+    prefs_table[key] = {
+      File = string.gsub(file_path.LuaString, "\\", "\\\\"),
+      Date =  os.date(),
+      FinaleVersion = finenv.FinaleVersion,
+      PluginVersion = finaleplugin.Version
+    }
+    if not transform_categories then 
+        prefs_table[key].Transformed = false 
+    end
+    if normalize_units then
+        prefs_table[key].DefaultUnit = "ev"
+    end
+end
+
+function options_save_as_json()
+    local documents = finale.FCDocuments()
+    documents:LoadAll()
+    local document = documents:FindCurrent()
+
+    local file_to_write = do_save_as_dialog(document)
+    if not file_to_write then
+        return
+    end
+
+    local file = io.open(file_to_write, "w")
```

Comment:
Likewise, we're starting to duplicate writing to file behavior across multiple scripts. Would be good to abstract this out at some point.

Reply:
I was thinking of that -- I basically stole the File Save and write to file from Robert's `document_save_as_text` :smile: